### ISSUE TITLE
Error preparing player items

### ISSUE THREAD

**Issue by RuiMarquesCardona** [jarnedemeulemeester/findroid#174]
Throws error preparing player items at some video files while others are fine, all files are mkv

 - Device: OnePlus7t Pro
 - Android version: 11
 - App version: 0.6.1(12)
 
null
 java.lang.NullPointerException
	at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.R(Unknown Source:99)
	at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.j0(Unknown Source:24)
	at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.Q(Unknown Source:116)
	at n7.j0.j(Unknown Source:16)
	at v8.a.k(Unknown Source:8)
	at k9.q0.run(Unknown Source:86)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:233)
	at android.app.ActivityThread.main(ActivityThread.java:8010)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)

**Comment by jarnedemeulemeester** [comment]
This is a weird issue. There are only a few reasons why there could be a `NullPointerException`:
- the `seriesId` of the item is null
- the `seasonId` of the item is null

Or maybe the `mediaSource` does not contain any `mediaStreams` (is null)

Somebody on the Discord server is haveing a simular issue.

This is the de-obfuscated stacktrace:
```
null
java.lang.NullPointerException
at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.episodeToPlayerItems(PlayerViewModel.java:99)
at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.prepareMediaPlayerItems(PlayerViewModel.java:24)
at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.createItems(PlayerViewModel.java:0)
at dev.jdtech.jellyfin.viewmodels.PlayerViewModel.access$createItems(PlayerViewModel.java:0)
at dev.jdtech.jellyfin.viewmodels.PlayerViewModel$createItems$1.invokeSuspend(PlayerViewModel.java:16)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:8)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:86)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:233)
at android.app.ActivityThread.main(ActivityThread.java:8010)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
```

**Comment by AnonymousWebHacker** [comment]
https://github.com/jarnedemeulemeester/findroid/issues/182#issuecomment-1288019403

**Comment by msycn** [comment]
add_header X-Frame-Options "SAMEORIGIN";
add_header X-XSS-Protection "1; mode=block";
add_header X-Content-Type-Options "nosniff";

**Comment by AnonymousWebHacker** [comment]
@msycn and @RuiMarquesCardona 

If you are the owner of the server, you should give details of the video that was tried to play.
Try the first outputs of ffplay video.mkv
It would be nice to know the format, etc.

Probably your error is with videos in h265 hevc at 10 bit

**Comment by msycn** [comment]
> 

Thanks,you are right👍, my videos format are 1080p.h265.10bit.mkv。 but my issue only on Linux server。when I try to use Windows server，it is so smoothly. So I think it is jellyfin server issue.

**Comment by jarnedemeulemeester** [comment]
The issue here has nothing to do with the codec of the video.

@RuiMarquesCardona what's the version of the Jellyfin server?

**Comment by AnonymousWebHacker** [comment]
it could be other simple issues like nginx configurations, or file permissions https://github.com/jarnedemeulemeester/findroid/issues/182#issuecomment-1292717535

@msycn  with jellyfin android/desktop client, do you play the videos well?

**Comment by msycn** [comment]
> it could be other simple issues like nginx configurations, or file permissions in server [#182 (comment)](https://github.com/jarnedemeulemeester/findroid/issues/182#issuecomment-1292717535)
> 
> @msycn with jellyfin android/desktop client, do you play the videos well?
 I can't play the videos with android/desktop client,so this server issues

**Comment by RuiMarquesCardona** [comment]
> The issue here has nothing to do with the codec of the video.
> 
> @RuiMarquesCardona what's the version of the Jellyfin server?

Hi, it's 10.8.4

Oddly enough, I installed Android 13 on my phone and it suddenly started working without any problems. I'll comment again if the issue comes back

**Comment by jarnedemeulemeester** [comment]
I will close this as this issue is gone for now.
If the problem comes back feel free to re-open this issue.

**Comment by donnykurnia** [comment]
Hi all. I just tried jellyfin and I encountered this bug. After tinkering about why this bug affects only some of the shows, I finally came to a conclusion. When I first add the sows to my jellyfin library, I don't see the shows in the shows list. So I read the jellyfin documentation, this page particularly: https://jellyfin.org/docs/general/server/media/shows

Then, I arrange the folder of my shows to match the folder structure in the documentation above. And then I add the `Shows` folder to the jellyfin library. This works well, in the official app and in findroid. But there is one of the shows not working in findroid, and I'm getting 'Error preparing player items'. Turns out, I add the folder named the shows itself, not the folder above it. This particular show is in a separate folder from the main 'Shows`. That's why only this one that not worked in findroid.

After I change the folder structure and re-adding the folder above the show itself, then findroid can play the media inside it.
For the illustration, here is the structure that  working:
```
Shows
├── Series Name (2010)
│   ├── Season 01
│   │   ├── Episode 01 file.mkv

```
In the jellyfin administration, the `Shows` folder that must be added to the library. But, if the folder `Series Name (2010)` that added to the library, this bug appear when trying to play the episode.
